examples/diffusion/example_diffusion.py

```diff
@@ -120,15 +120,18 @@ def execute_devito(ui, spacing=0.01, a=0.5, timesteps=500):
     # Note: This should be made simpler through the use of defaults
     u = TimeData(name='u', shape=(nx, ny), time_order=1, space_order=2)
     u.data[0, :] = ui[:]
+    u.indices[0].size = timesteps
+    u.indices[1].size = nx
+    u.indices[2].size = ny
 
     # Derive the stencil according to devito conventions
     eqn = Eq(u.dt, a * (u.dx2 + u.dy2))
     stencil = solve(eqn, u.forward)[0]
-    op = Operator(stencils=Eq(u.forward, stencil), subs={h: spacing, s: dt},
-                  nt=timesteps, shape=(nx, ny), spc_border=1, time_order=1)
+    op = StencilKernel(Eq(u.forward, stencil), subs={h: spacing, s: dt})
+
     # Execute the generated Devito stencil operator
     tstart = time.time()
-    op.apply()
+    op.apply(u)
```

Comment:
This is diffusion, so there are no `m` or `damp` terms. The core change here though is that the kernel accepts all numpy arrays that have the same dimension as `u.data` and is not bound to the symbol itself anymore.